homevork/07-mars-viewer/src/redux/marsImagesSlice.ts

```diff
@@ -0,0 +1,109 @@
+import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import {IMarsImage} from '../interfaces/marsImage';
+import {RootState} from './store';
+
+export enum Status {
+  Loading = 'Loading',
+  NotLoaded = 'NotLoaded',
+  Loaded = 'Loaded',
+  Failed = 'Failed',
+  EmptyArray = 'EmptyArray',
+
+}
+
+type Images = {
+  [key: string]: IMarsImage[];
+};
+
+type InitialState = {
+  images: Images,
+  favoriteImages:IMarsImage[],
+  solSelected:number,
+  status:Status,
+};
+
+type ServerResponse = {
+  photos:{
+    id:number,
+    img_src:string,
+    rover:{name:string}
+    camera:{full_name:string}
+  }[]
+};
+
+const initialState:InitialState= {
+  images: {},
+  favoriteImages: [],
+  solSelected: 0,
+  status: Status.NotLoaded,
+};
+
+export const fetchImages:any = createAsyncThunk<IMarsImage[], number, {}>(
+  'marsImages/fetchMarsImages',
+  async (sol:number) => {
+    const res = await fetch(
+      `https://api.nasa.gov/mars-photos/api/v1/rovers/perseverance/photos?sol=${sol}&page=2&api_key=gB92ijgzKbffEy2HfTNDlrvGbCP97NofvGwuqUlJ`,
+      {
+        method: 'GET',
+      },
+    );
+
+    const data:ServerResponse = await res.json();
+    const mappedData:IMarsImage[] = data.photos.map((image) => ({
+      id: image.id,
+      img_src: image.img_src,
+      rover: image.rover.name,
+      camera: image.camera.full_name,
+      isFavorite: false,
+    }));
+
+    return mappedData;
+  },
+);
+
+export const marsImagesSlice = createSlice({
+  name: 'marsImages',
+  initialState,
+  reducers: {
+    addToFavorites: (state, action:PayloadAction<IMarsImage>) => {
+      const index = state.images[state.solSelected].findIndex((image) => image.id === action.payload.id);
+      state.images[state.solSelected][index].isFavorite = true;
+      state.favoriteImages.push(state.images[state.solSelected][index]);
```

Comment:
state duplication, it either should be in list of favorites, or it has flag `isFavorite` imagine if some business logic adds to this app and you forget to update this in both places, it will lead to state being inconsistent and later to ui bugs